packages/invoices/src/OutboundShipment/api/hooks.ts

```diff
@@ -141,3 +146,92 @@ export const useSaveOutboundLines = () => {
     },
   });
 };
+
+export const useDeleteInboundLine = (): UseMutationResult<
+  DeleteOutboundShipmentLinesMutation,
+  unknown,
+  string[],
+  { previous?: Invoice; ids: string[] }
+> => {
+  // TODO: Shouldn't need to get the invoice ID here from the params as the mutation
+  // input object should not require the invoice ID. Waiting for an API change.
+  const { id = '' } = useParams();
+  const queryClient = useQueryClient();
+  const { api } = useOmSupplyApi();
+  const mutation = OutboundApi.deleteLines(api, id);
+  return useMutation(mutation, {
+    onMutate: async (ids: string[]) => {
+      await queryClient.cancelQueries(['invoice', id]);
+
+      const previous = queryClient.getQueryData<Invoice>(['invoice', id]);
+
+      if (previous) {
+        queryClient.setQueryData<Invoice>(['invoice', id], {
+          ...previous,
+          lines: previous.lines.filter(
+            ({ id: lineId }) => !ids.includes(lineId)
+          ),
+        });
+      }
+
+      return { previous, ids };
+    },
+    onError: (_, __, context) => {
+      queryClient.setQueryData(['invoice', id], context?.previous);
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries(['invoice', id]);
+    },
+  });
+};
```

Comment:
this is the actual mutation hook, which is the same as inbound, optimistically deleting